source/common/upstream/load_balancer_impl.cc

```diff
@@ -714,20 +733,37 @@ void EdfLoadBalancerBase::initialize() {
   }
 }
 
+void EdfLoadBalancerBase::recalculateHostsInSlowStart(const HostVector& hosts) {
+  auto current_time = time_source_.monotonicTime();
+  for (const auto& host : hosts) {
+    auto host_create_duration =
+        std::chrono::duration_cast<std::chrono::milliseconds>(current_time - host->creationTime());
+    // Check if host existence time is within slow start window.
+    if (host->creationTime() > latest_host_added_time_ &&
+        host_create_duration <= slow_start_window_ &&
+        host->health() == Upstream::Host::Health::Healthy) {
+      latest_host_added_time_ = host->creationTime();
+    }
+  }
```

Comment:
We decided to fall back to simpler version of slow start algorithm to reduce code complexity. In previous version there was an ordered hash map struct with constant lookup that also required compaction with time (to purge stale hosts). I think is fine for initial version and would like to revisit it in case there is a perf penalty. Added a TODO.

Reply:
We have agreed to switch to simpler version of slow start algorithm to reduce code complexity. The previous version was maintaining ordered hash map to track hosts in slow start and required compaction as time progressed (to purge stale hosts). I think is fine to have this unoptimised linear scan in initial version, but would like to revisit in case there is a noticeable perf penalty. Added TODO